stubs/CoreGenericFunctions.phpstub

```diff
@@ -661,6 +696,103 @@ function htmlspecialchars_decode(string $string, ?int $quote_style = null) : str
  */
 function str_replace($search, $replace, $subject, &$count = null) {}
 
+/**
+ * @psalm-pure
+ *
+ * @param string|array<string|int|float> $search
+ * @param string|array<string|int|float> $replace
+ * @param string|array<string|int|float> $subject
+ * @param int $count
+ * @return ($subject is array ? array<string> : string)
+ *
+ * @psalm-flow ($replace, $subject) -> return
+ */
+function str_ireplace($search, $replace, $subject, &$count = null) {}
+
+/**
+ * @psalm-pure
+ *
+ * @psalm-flow ($input, $pad_string) -> return
+ */
+function str_pad(string $input, int $pad_length, $pad_string = '', int $pad_type = STR_PAD_RIGHT): string {}
+
+/**
+ * @psalm-pure
+ *
+ * @psalm-flow ($input) -> return
+ */
+function str_repeat(string $input, int $multiplier): string {}
+
+/**
+ * @psalm-pure
+ *
+ * @psalm-flow ($str) -> return
+ */
+function str_rot13(string $str): string {}
+
+/**
+ * @psalm-pure
+ *
+ * @psalm-flow ($str) -> return
+ */
+function str_shuffle(string $str): string {}
+
+/**
+ * @psalm-pure
+ *
+ * @psalm-flow ($string) -> return
+ */
+function str_split(string $string, int $split_length = 1): array {}
```

Comment:
I guess this should be more specific considering the test failure seems related to https://github.com/vimeo/psalm/blob/ebf33723cb2128cb2f34bb9cd87321c661e18fac/src/Psalm/Internal/FileManipulation/FunctionDocblockManipulator.php#L160-L161